src/buildblock/SeparableMetzArrayFilter.cxx

```diff
@@ -240,38 +184,29 @@ void build_metz(VectorWithOffset<elemT>& kernel,
     
     /* Build the fft array, odd coefficients are the imaginary part */
     
-    for (i=0;i<=Res-(Res/2);i++) {
-      fftdata[REALC(i)]=filter[Res/2-1+i];
-      fftdata[IMGC(i)] = 0.0;
+    for (int i=0;i<=Res-(Res/2);i++) {
+      fftdata[i].real(filter[Res/2-1+i]);
     }
     
-    for (i=1;i<(Res/2);i++) {
-      fftdata[REALC(Res-(Res/2)+i)]=filter[i-1];
-      fftdata[IMGC(Res-(Res/2)+i)] = 0.0;
+    for (int i=1;i<(Res/2);i++) {
+      fftdata[Res-(Res/2)+i].real(filter[i-1]);
     }
-    
-    
+
     /* FFT to frequency space */
-    fftdata.set_offset(1);
-    discrete_fourier_transform(fftdata/*-1*/,Res,FORWARDFFT); 
-    fftdata.set_offset(0);
-    
-    
-    
+    fourier< stir::Array<1,std::complex<elemT> > >(fftdata);
```

Comment:
do you actually need the explicit type? I didn't think so

Reply:
Indeed. I thought it was more readable. 
:ok: